app/scripts/controllers/mmi-controller.js

```diff
@@ -39,7 +38,12 @@ export default class MMIController extends EventEmitter {
     this.signatureController = opts.signatureController;
     this.platform = opts.platform;
     this.extension = opts.extension;
-    this.trackTransactionEvents = opts.trackTransactionEvents;
+
+    const { hooks } = opts ?? {};
+    this.getTransactions = hooks?.getTransactions;
+    this.updateTransactionHash = hooks?.updateTransactionHash;
+    this.trackTransactionEvents = hooks?.trackTransactionEvents;
```

Comment:
This might be the quickest and best fix for now, but ideally this controller wouldn't be coupled to metrics and would instead emit standard events that the `MetaMaskController` could listen to and create the necessary metrics, as we now do in the `TransactionController`.

Probably best as a future ticket though given size and risk.